internal/loader/keyvault_reference_resolver.go

```diff
@@ -28,6 +28,7 @@ type KeyVaultSecretUriSegment struct {
 	HostName      string
```

Comment:
This is longer just indicates the UriSegment of the Secret, how about rename this type to `KeyVaultSecretMetadata` ?